portal/lib/doc_lib.php

```diff
@@ -71,52 +73,41 @@
     if ($_SESSION['language_direction'] == 'rtl') {
         $pdf->SetDirectionality('rtl');
     }
+
+    $htmlin = "<html><body>$htmlin</body></html>";
+    // need custom stylesheet for templates
     $pdf->writeHtml($htmlin);
+
     if ($dispose == 'download') {
         header('Content-type: application/pdf');
-        header('Content-Disposition: attachment; filename=$form_filename');
+        header("Content-Disposition: attachment; filename=$form_filename");
         $pdf->Output($form_filename, 'D');
-        $logit->portalLog('download document', $cpid, ('document:'.$form_filename));
+        $logit->portalLog('download document', $cpid, ('document:' . $form_filename));
+        echo js_escape('okay');
```

Comment:
What are all the js_escape calls in this script supposed to be doing?
(also, are then in js context or html context?)